### PR TITLE
net: lib: nrf_cloud_coap: Extend proprietary PSM support

### PR DESCRIPTION
**Pull request by plskeggs**
Select LTE proprietary PSM mode for the 9131 and 9151 SOCs in addition to the existing 9161 selection, when building for CoAP connectivity to nRF Cloud.

Jira: IRIS-8450

### CODE REVIEW

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -342,7 +342,9 @@ Modem libraries
 Libraries for networking
 ------------------------
 
-|no_changes_yet_note|
+* :ref:`lib_nrf_cloud_coap` library:
+
+  * Updated to request proprietary PSM mode for SOC_NRF9151_LACA and SOC_NRF9131_LACA in addition to SOC_NRF9151_LACA || SOC_NRF9161_LACA.
```
Comment (jayteemo) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:347]:
```suggestion
  * Updated to request proprietary PSM mode for SOC_NRF9151_LACA and SOC_NRF9131_LACA in addition to SOC_NRF9161_LACA.
```

**Review by jayteemo — APPROVED** [review]

**Review by glarsennordic — APPROVED** [review]
Point of curiosity, why do we enable this specifically for CoAP?

**Comment by plskeggs** [comment]
> Point of curiosity, why do we enable this specifically for CoAP?

Our CoAP service does not send unsolicited packets to the device, so it is safe to use proprietary PSM mode, which keeps the receiver off while in power saving mode.  Regular PSM keeps the receiver on, so unsolicited packets will be received.

**Comment by glarsennordic** [comment]
Ahhh, thank you, that makes sense

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -342,7 +342,9 @@ Modem libraries
 Libraries for networking
 ------------------------
 
-|no_changes_yet_note|
+* :ref:`lib_nrf_cloud_coap` library:
+
+  * Updated to request proprietary PSM mode for SOC_NRF9151_LACA and SOC_NRF9131_LACA in addition to SOC_NRF9161_LACA.
```
Comment (peknis) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:347]:
```suggestion
  * Updated to request proprietary PSM mode for ``SOC_NRF9151_LACA`` and ``SOC_NRF9131_LACA`` in addition to ``SOC_NRF9161_LACA``.
```

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by peknis — APPROVED** [review]

**Pull request merged**